dotnet/src/Experimental/Process.LocalRuntime/LocalProcess.cs

```diff
@@ -333,13 +351,27 @@ private void EnqueueExternalMessages(Queue<ProcessMessage> messageChannel)
     private void EnqueueStepMessages(LocalStep step, Queue<ProcessMessage> messageChannel)
     {
         var allStepEvents = step.GetAllEvents();
+        // alternative solution start
+        bool eventsChanged = this.EmitPublicEvents(allStepEvents);
+
+        if (eventsChanged)
+        {
+            var unprocessedEvents = allStepEvents.Where(stepEvent => stepEvent.Visibility != KernelProcessEventVisibility.Public).ToList();
```

Comment:
This might be a concatenated enumeration and avoid reallocating a collection.

Reply:
removed this code